### ISSUE TITLE
Exception in thread "main" ai.djl.translate.TranslateException: java.lang.IllegalArgumentException: Input number mismatch, requires: [im_shape, image, scale_factor]

### ISSUE THREAD

**Issue by JiMa98** [deepjavalibrary/djl#2376]
## Description
I try to export Yolov3/FastRCNN trained by paddlex to DJL. However, when the model is loaded by DJL and prepare to do inference task, DJL says these two models should have three input names. 

### Expected Behavior
DJL produces right DetectedObjects object by predict method.

### Error Message
Exception in thread "main" ai.djl.translate.TranslateException: java.lang.IllegalArgumentException: Input number mismatch, requires: [im_shape, image, scale_factor]
	at ai.djl.inference.Predictor.batchPredict(Predictor.java:189)
	at ai.djl.inference.Predictor.predict(Predictor.java:126)
	at FaceDetection.main(FaceDetection.java:67)
Caused by: java.lang.IllegalArgumentException: Input number mismatch, requires: [im_shape, image, scale_factor]
	at ai.djl.paddlepaddle.engine.PpSymbolBlock.forwardInternal(PpSymbolBlock.java:56)
	at ai.djl.nn.AbstractBaseBlock.forward(AbstractBaseBlock.java:77)
	at ai.djl.nn.Block.forward(Block.java:123)
	at ai.djl.inference.Predictor.predictInternal(Predictor.java:140)
	at ai.djl.inference.Predictor.batchPredict(Predictor.java:167)
	... 2 more

## How to Reproduce?
The YOLOv3 paddlex demo I use is : https://aistudio.baidu.com/aistudio/projectdetail/2160238
The export command I use is :  !paddlex --export_inference --model_dir=./output/yolov3_darknet53/best_model/ --save_dir=./yolov3

The FastRCNN paddlex demo I use is : https://aistudio.baidu.com/aistudio/projectdetail/2160245
The export command I use is :  !paddlex --export_inference --model_dir=./output/faster_rcnn_r50_fpn/best_model/ --save_dir=./RCNN

The version of paddlex I used is 2.0.0

After I get the model zip file, I rename 3 files of 5 in the zip (change model.pdmodel to inference.pdmodel / model.pdiparams to inference.pdiparams / model.pdiparams.info to inference.pdiparams.info).

Then I start to write DJL code as below:

DetectionDemo.java:

        String url = "C:\\Users\\mj\\Desktop\\djldemo1\\src\\main\\resources\\0013.jpg";
        Image img = ImageFactory.getInstance().fromFile(Paths.get(url));
        Criteria<Image, DetectedObjects> criteria = Criteria.builder()
                        .optEngine("PaddlePaddle")
                        .setTypes(Image.class, DetectedObjects.class)
                        .optModelUrls("file:/C:/Users/mj/Desktop/djldemo1/model/RCNN.zip")
                        .optTranslator(new FaceTranslator(0.5F,0.7F))
                        .optProgress(new ProgressBar())
                        .build();

        var model = criteria.loadModel();
        var predictor = model.newPredictor();

        DetectedObjects inferenceResult = predictor.predict(img);

FaceTranslator.java (Almost the same as the demo on the website : http://docs.djl.ai/jupyter/paddlepaddle/face_mask_detection_paddlepaddle_zh.html) except the "className":

class FaceTranslator implements NoBatchifyTranslator<Image, DetectedObjects> {

    private float shrink;
    private float threshold;
    private List<String> className;

    FaceTranslator(float shrink, float threshold) {
        this.shrink = shrink;
        this.threshold = threshold;
        className = Arrays.asList("leconte",
                "boerner",
                "armandi",
                "linnaeus",
                "coleoptera",
                "acuminatus");
    }

    @Override
    public DetectedObjects processOutput(TranslatorContext ctx, NDList list) {
        return processImageOutput(list, className, threshold);
    }

    @Override
    public NDList processInput(TranslatorContext ctx, Image input) {
        return processImageInput(ctx.getNDManager(), input, shrink);
    }

    NDList processImageInput(NDManager manager, Image input, float shrink) {
        NDArray array = input.toNDArray(manager);
        Shape shape = array.getShape();
        System.out.println(shape);
        array = NDImageUtils.resize(
                array, (int) (608), (int) (608));
        array = array.transpose(2, 0, 1).flip(0); // HWC -> CHW BGR -> RGB
        NDArray mean = manager.create(new float[] {104f, 117f, 123f}, new Shape(3, 1, 1));
        array = array.sub(mean).mul(0.007843f); // normalization
        array = array.expandDims(0); // make batch dimension

        return new NDList(array);
    }

    DetectedObjects processImageOutput(NDList list, List<String> className, float threshold) {
        NDArray result = list.singletonOrThrow();
        float[] probabilities = result.get(":,1").toFloatArray();
        List<String> names = new ArrayList<>();
        List<Double> prob = new ArrayList<>();
        List<BoundingBox> boxes = new ArrayList<>();
        for (int i = 0; i < probabilities.length; i++) {
            if (probabilities[i] >= threshold) {
                float[] array = result.get(i).toFloatArray();
                names.add(className.get((int) array[0]));
                prob.add((double) probabilities[i]);
                boxes.add(
                        new Rectangle(
                                array[2], array[3], array[4] - array[2], array[5] - array[3]));
            }
        }
        return new DetectedObjects(names, prob, boxes);
    }
}

TEST image:
![0013](https://user-images.githubusercontent.com/60089539/217563616-ec6dccd4-f885-48af-bdfa-c416833db38e.jpg)

## What have you tried to solve it?

1. I tried the classification model trained by paddlex in DJL and everything works.
2. I tried the modelzoo's model and the model's link in the DJL tutorial.(Such as OCR detection and face detection), everything works.
3. I found one similar error message when someone tried to convert YOLOv3 paddle model to openvino(https://github.com/openvinotoolkit/openvino/issues/9109)
4. I found the exception comes from (PpSymbolBlock.java:56), inputs.size() == 1, inputNames.length == 3. When I observe other successful demos, I found that the inputNames‘ length always equals to 1. But I don't know what happend during "JniUtils.getInputNames(predictor)" in (PpSymbolBlock.java:44)

## Environment Info

java 11

paddlex version 2.0.0

os window10

train with V100

inference with CPU

djl version 0.19.0

dependecies:
     <dependency>
            <groupId>ai.djl</groupId>
            <artifactId>api</artifactId>
            <version>${djl.version}</version>
        </dependency>

        
        <dependency>
            <groupId>tech.tablesaw</groupId>
            <artifactId>tablesaw-jsplot</artifactId>
            <version>0.38.1</version>
        </dependency>
        
        <dependency>
            <groupId>org.apache.commons</groupId>
            <artifactId>commons-lang3</artifactId>
            <version>3.12.0</version>
        </dependency>

        <dependency>
            <groupId>org.slf4j</groupId>
            <artifactId>slf4j-simple</artifactId>
            <version>1.7.26</version>
        </dependency>
        <dependency>
            <groupId>ai.djl.paddlepaddle</groupId>
            <artifactId>paddlepaddle-model-zoo</artifactId>
            <version>${djl.version}</version>
        </dependency>

        <dependency>
            <groupId>ai.djl.pytorch</groupId>
            <artifactId>pytorch-engine</artifactId>
            <version>${djl.version}</version>
        </dependency>
        <dependency>
            <groupId>ai.djl</groupId>
            <artifactId>basicdataset</artifactId>
            <version>${djl.version}</version>
        </dependency>

**Comment by JiMa98** [comment]
@lanking520 @frankfliu  Help please.

**Comment by frankfliu** [comment]
@JiMa98 
Your model expect 3 inputs (NDList should have 3 NDArray), but your translator only adds image to the NDList.
You should add image shape, image, and scale factor into NDList

**Comment by JiMa98** [comment]
> 

Rhank you ! But what should I write to satisfy that requirement(specific formats or code demo please~). Also, I noticed other object demos using Yolo trained by paddleDetection doesn't require 3 inputs. They only return new NDList(array) in the processInput function. Could you please give me some instructions about training model using Paddle or Paddlex, just like what you have did in the FaceDetection demo. You give the Model URL directly so that I don't know the right way to train a model like that. @frankfliu

**Comment by JiMa98** [comment]
> @JiMa98 Your model expect 3 inputs (NDList should have 3 NDArray), but your translator only adds image to the NDList. You should add image shape, image, and scale factor into NDList

I just using the demo in the paddlex. There isn‘t something special in the model. The pretrained model is gotten directly from the paddlex's model zoo. Also, when using paddlex for prediction, I only need to give the image itself for input without the need of 3 inputs. So I actually don't know where 3 inputs comes from and how I should write them....

**Comment by JiMa98** [comment]
> 

Thank you！ I have fixed the problem. The input format is defined by Paddle. 
I used java codes like this:

    NDList processImageInput(NDManager manager, Image input) {

        NDArray array = input.toNDArray(manager);
        Shape shape = array.getShape();

        array = NDImageUtils.resize(
                array, (int) (416), (int) (416));
        array = array.transpose(2, 0, 1).flip(0); // HWC -> CHW BGR -> RGB
        NDArray mean = manager.create(new float[] {104f, 117f, 123f}, new Shape(3, 1, 1));
        array = array.sub(mean).mul(0.007843f); // normalization
        array = array.expandDims(0); // make batch dimension

        NDArray scale_factor = manager.create(new float[]{416f/shape.get(0),416f/shape.get(1)}).reshape(1,2);

        NDArray im_shape = manager.create(new float[]{416,416}).reshape(1,2);


        return new NDList(im_shape,array,scale_factor);

    }
